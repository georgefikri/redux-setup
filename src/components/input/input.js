import React, { forwardRef } from 'react';
import { capitalizeFirstWord } from '../../helpers';
import './input.scss';
const input = forwardRef(
  (
    { placeholder, value, id, name, iconsBoolean, icons, classes, type, onChange, disabled },
    ref
  ) => {
    return (
      <div>
        <div className="input-field-container">
          {iconsBoolean && (
            <span className="input-icon-cont">
              <span className={icons} />
            </span>
          )}
          <input
            type={type}
            ref={ref}
            className={classes}
            id={id}
            // data-testid={id}
            placeholder={capitalizeFirstWord(placeholder)}
            value={value}
            name={name}
            // eslint-disable-next-line no-console
            onChange={(e) => onChange(e)}
            disabled={disabled}
            data-testid='input'

          />
        </div>
      </div>
    );
  }
);
export const Input = input;
