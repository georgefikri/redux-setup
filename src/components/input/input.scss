@import '../../styles/variables.scss';

.input-field-container {
  .input-icon-cont {
    height: 36px;
    background-color: $orance-color;
    width: 34px;
    display: inline-block;
    color: $white-color;
    text-align: center;
    font-size: $input-icons-size;
    position: absolute;
    padding-top: 5px;
    margin-top: 1px;
  }
}

.form-horizontal {
  .form-input-missing,
  .multi-select-missing > div div,
  [class*='-missing'] {
    border-color: $grey-color !important;
  }
}

.vertical-tabs-rightCont .input-field-container input {
  padding-left: 40px;
}
