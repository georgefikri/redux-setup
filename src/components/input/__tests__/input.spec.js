import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
// import { HashRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Input } from '../input';

let onChange = jest.fn();

describe('testing delayed input', ()=>{
  afterEach(cleanup);


  test('should write in input', () => {
    const { getByTestId } = render(<Input data-testid='input' onChange={()=> onChange()}/>);
    const input = getByTestId('input');
    fireEvent.change(input, { target: { value: '23' } });
    expect(input.value).toBe('23');
  });

  //   snapshot
  test('should do snapshot', () => {
    const input = renderer.create(
      <Input
        onChange={()=> onChange()}
      />
    ).toJSON();
    expect(input).toMatchSnapshot();
  });

});