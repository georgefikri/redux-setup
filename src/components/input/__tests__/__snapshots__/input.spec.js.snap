// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing delayed input should do snapshot 1`] = `
<div>
  <div
    className="input-field-container"
  >
    <input
      data-testid="input"
      onChange={[Function]}
    />
  </div>
</div>
`;
